sql/core/src/test/scala/org/apache/spark/sql/ColumnExpressionSuite.scala

```diff
@@ -331,4 +331,186 @@ class ColumnExpressionSuite extends QueryTest {
     assert(schema("value").metadata === Metadata.empty)
     assert(schema("abc").metadata === metadata)
   }
+
+  def testOneToOneMathFunction[@specialized(Int, Double, Float, Long) T]
```

Comment:
I'd create a new suite for all the math functions, since there are so many of them. 

No need for specialization, is there?